### PR TITLE
Options refactor

### PR DESCRIPTION
**Pull request by jmaslek**
No new features added.

Refactored so that argparse logic in controller and model/view decoupled

### CODE REVIEW

File: gamestonk_terminal/options/tradier_view.py [hunk 1]
```diff
@@ -419,23 +403,39 @@ def plot_vol(
...
+    export: str
+        Format for exporting data
     """
-
     current_price = tradier_model.last_price(ticker)
+    options = tradier_model.get_option_chains(ticker, expiry)
+    export_data(
+        export,
+        os.path.dirname(os.path.abspath(__file__)),
+        "tr_voi",
```
Comment (DidierRLopes) [inline gamestonk_terminal/options/tradier_view.py:435]: can we have this the other way around `command(source)`

File: gamestonk_terminal/options/tradier_view.py [hunk 2]
```diff
@@ -591,115 +591,73 @@ def plot_volume_open_interest(
...
-        default=False,
-        help="Flag for showing put option",
-    )
-    parser.add_argument("--chain", dest="chain_id", type=str, help="OCC option symbol")
 
-    parser.add_argument(
-        "--raw", dest="raw", action="store_true", default=False, help="Display raw data"
-    )
+    if raw:
+        print(df_hist)
```
Comment (DidierRLopes) [inline gamestonk_terminal/options/tradier_view.py:628]: Should this be tabulate?

File: gamestonk_terminal/options/options_controller.py [hunk 3]
```diff
@@ -208,102 +306,762 @@ def call_unu(self, other_args: List[str]):
...
+            "--put",
+            dest="put",
+            action="store_true",
+            default=False,
+            help="Flag for showing put option",
+        )
+        parser.add_argument(
+            "--chain", dest="chain_id", type=str, help="OCC option symbol"
+        )
+
```
[the remark this answers is not in the document]
Comment (jmaslek) [reply gamestonk_terminal/options/options_controller.py:664]: I don't get it.  Like get rid of space below?
```python
parser.add(...)
space
parser.add(...)
```

File: gamestonk_terminal/options/options_controller.py [hunk 4]
```diff
@@ -208,102 +306,762 @@ def call_unu(self, other_args: List[str]):
...
     def call_exp(self, other_args: List[str]):
         """Process exp command"""
-        if self.ticker:
-            self.selected_date = op_helpers.select_option_date(
-                self.expiry_dates, other_args
-            )
-        else:
-            print("Please select a ticker using load {ticker}", "\n")
+
```
[the remark this answers is not in the document]
Comment (jmaslek) [reply gamestonk_terminal/options/options_controller.py:575]: you need to select more lines in the comments lol.  I have no idea where this spacing is

[the remark this answers is not in the document]
Comment (jmaslek) [reply gamestonk_terminal/options/options_controller.py:575]: you need to select more lines in the comments lol.  I have no idea where this spacing is

File: gamestonk_terminal/options/options_controller.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (DidierRLopes) [reply gamestonk_terminal/options/options_controller.py:664]: yes, exactly that. We can add a new line when it's a new if, or a new "group of stuff". But for adding multiple possible vars, I'd rather no spacing

File: gamestonk_terminal/options/syncretism_view.py [hunk 5]
```diff
@@ -1,539 +1,174 @@
...
-            raise argparse.ArgumentTypeError("Invalid option chains header selected!")
+    export_data(
+        export,
+        os.path.dirname(os.path.abspath(__file__)),
+        "scr",
+        df_res,
+    )
 
-    return columns
+    if n_show > 0:
```
[the remark this answers is not in the document]
Comment (jmaslek) [reply gamestonk_terminal/options/syncretism_view.py:96]: So the reason I have this is that default n_show is -1.  With check_positive, would it just be `if n_show`

File: gamestonk_terminal/options/options_controller.py [hunk 6]
```diff
@@ -208,102 +306,762 @@ def call_unu(self, other_args: List[str]):
...
+            formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+            prog="exp",
+            description="See and set expiration date",
+        )
+        parser.add_argument(
+            "-d",
+            "--date",
+            dest="n_date",
+            action="store",
+            type=int,
```
[the remark this answers is not in the document]
Comment (jmaslek) [reply gamestonk_terminal/options/options_controller.py:587]: that raises an error with 0, which is a valid choice

[the remark this answers is not in the document]
Comment (DidierRLopes) [reply gamestonk_terminal/options/options_controller.py:587]: you have check_non_negative then :b

File: gamestonk_terminal/options/syncretism_view.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (DidierRLopes) [reply gamestonk_terminal/options/syncretism_view.py:96]: Yea, true. Leave it like this! I didn't realize you were showing it anyway, just sampling it if there is a value chosen

**Review by DidierRLopes — APPROVED** [review]

**Pull request merged**